### PR TITLE
[PORT] build: python 3 compatibility

### LINKED ISSUE #1395: dev HEAD: error on `make deploy` when using Python 3.6
```
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/Resources
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/Resources
/opt/local/bin/ginstall -c -m 644 src/qt/res/icons/bitcoin.icns Bitcoin-Qt.app/Contents/Resources/bitcoin.icns
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents
/opt/local/bin/ginstall -c -m 644 share/qt/Info.plist Bitcoin-Qt.app/Contents/Info.plist
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src qt/bitcoin-qt
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/MacOS
STRIPPROG="/opt/local/bin/strip" /bin/sh /Users/rex/BitcoinUnlimited/build-aux/install-sh -c -s  src/qt/bitcoin-qt Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt
/opt/local/bin/gmkdir -p Bitcoin-Qt.app/Contents/Resources/Base.lproj
echo '{	CFBundleDisplayName = "Bitcoin Unlimited Cash Edition"; CFBundleName = "Bitcoin Unlimited Cash Edition"; }' > Bitcoin-Qt.app/Contents/Resources/Base.lproj/InfoPlist.strings
./contrib/macdeploy/macdeployqtplus Bitcoin-Qt.app -add-qt-tr da,de,es,hu,ru,uk,zh_CN,zh_TW -translations-dir= -dmg -fancy ./contrib/macdeploy/fancy.plist -verbose 2 -volname Bitcoin-Unlimited-Cash-Edition
  File "./contrib/macdeploy/macdeployqtplus", line 204
    print "Inspecting with otool: " + binaryPath
                                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Inspecting with otool: " + binaryPath)?
make: *** [Bitcoin-Unlimited-Cash-Edition.dmg] Error 1
```

### PR DESCRIPTION
**Pull request by sickpig**
This is a port of bitcoin/bitcoin/pull/7723 that fix #1395 

Ubuntu 16.04 "xenial xerus" does not come with Python 2.x by default.
It is possible to install a python-2.7 package, but this has its own
problem: no `python` or `python2` symlink (see #7717).

This fixes the following scripts to work with python 3:
- `make check` (bctest,py, bitcoin-util-test.py)
- `make translate` (extract_strings_qt.py)
- `make symbols-check` (symbol-check.py)
- `make security-check` (security-check.py)

Explicitly call the python commands using $(PYTHON) instead
of relying on the interpreter line at the top of the scripts.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -92,7 +92,8 @@ AC_PATH_TOOL(RANLIB, ranlib)
 AC_PATH_TOOL(STRIP, strip)
 AC_PATH_TOOL(GCOV, gcov)
 AC_PATH_PROG(LCOV, lcov)
-AC_PATH_PROG(PYTHON, python)
+AC_PATH_PROG(JAVA, java)
```
Comment (gubatron) [inline configure.ac]: why java on these set of changes?

Comment (sickpig) [reply configure.ac]: ouch it slips in. going to remove it. thanks!

**Comment by sickpig** [comment]
fixed

**Comment by awemany** [comment]
So IIUC, the `from __future__` imports are because the scripts still maintain `py2` compatibility? This is not really a nit even, but I do wonder whether it would make sense to drop `py2` support altogether to avoid the extra noise in the scripts.

**Comment by sickpig** [comment]
@awemany guess you are right. I'm going to remove any reference to py2 in a following PR

**Pull request merged**